spec/display-layer-spec.coffee

```diff
@@ -33,6 +34,39 @@ describe "DisplayLayer", ->
       expect(displayLayer2.ratioForCharacter).toBe(displayLayer1.ratioForCharacter)
       expect(displayLayer2.isWrapBoundary).toBe(displayLayer1.isWrapBoundary)
 
+  it "does not recompute the display layer's screen lines until they are needed", ->
+    spliceSpy = spyOn(DisplayIndex.prototype, 'splice').and.callThrough()
+    buffer = new TextBuffer(text: '\tif (a) { b; } else { c; }')
+
+    # construction does not force screen line computation
+    displayLayer = buffer.addDisplayLayer(tabLength: 4)
+    expect(spliceSpy.calls.count()).toBe(0)
```

Comment:
I think I am 👍 on spying on `splice`. Just as an alternative, we might consider querying the state of the `DisplayIndex` and avoid the spy:

```js
expect(displayLayer.displayIndex.getScreenLineCount()).toBe(1)
```